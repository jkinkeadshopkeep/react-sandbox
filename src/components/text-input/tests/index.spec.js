import React from 'react';
import { render } from '@testing-library/react';
import TextInput from '..';

jest.mock('../inputWrapper', () => 'mock-input-wrapper');

describe('Text input component', () => {
  const createComponent = (props) => <TextInput {...props} />;
  const defaultProps = {
    name: 'text-input-1',
    label: 'Your Name'
  };

  describe('when text input with no error is instantiated', () => {
    it('renders the text input', () => {
      const component = render(createComponent(defaultProps));
      expect(component.asFragment()).toMatchSnapshot();
    });
  });

  describe('when email input with no error is instantiated', () => {
    it('renders the email input', () => {
      const component = render(
        createComponent({
          ...defaultProps,
          type: 'email'
        })
      );
      expect(component.asFragment()).toMatchSnapshot();
    });
  });

  describe('when telephone input with no error is instantiated', () => {
    it('renders the telephone input', () => {
      const component = render(
        createComponent({
          ...defaultProps,
          type: 'tel'
        })
      );
      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
