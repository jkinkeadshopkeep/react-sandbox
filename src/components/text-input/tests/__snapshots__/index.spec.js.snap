// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text input component when email input with no error is instantiated renders the email input 1`] = `
<DocumentFragment>
  <mock-input-wrapper
    id="text-input-1"
    label="Your Name"
  >
    <input
      name="text-input-1"
      type="email"
    />
  </mock-input-wrapper>
</DocumentFragment>
`;

exports[`Text input component when telephone input with no error is instantiated renders the telephone input 1`] = `
<DocumentFragment>
  <mock-input-wrapper
    id="text-input-1"
    label="Your Name"
  >
    <input
      name="text-input-1"
      type="tel"
    />
  </mock-input-wrapper>
</DocumentFragment>
`;

exports[`Text input component when text input with no error is instantiated renders the text input 1`] = `
<DocumentFragment>
  <mock-input-wrapper
    id="text-input-1"
    label="Your Name"
  >
    <input
      name="text-input-1"
      type="text"
    />
  </mock-input-wrapper>
</DocumentFragment>
`;
